### ISSUE TITLE
Nested DnD & Circular Dependency

### ISSUE THREAD

**Issue by mnashmi** [isaacHagoel/svelte-dnd-action#307]
Hiya,

Im trying to create recursive/nested drag and drop zones based on a config json.

When moving an entire zone along with its children in a parent node, i get the following errors:
1- Cannot destructure property 'items' of 'dzToConfig.get(...)' as it is undefined.
2- Uncaught (in promise): Maximum call stack size exceeded

Im guessing its the way im approaching it as the maximum call stack is a screaming indicator, but not sure how to fix it.

The idea is, you have a root drag and drop container with a list of blocks (buttons, text, etc.) A block can also be a drag and drop container (ie. a list) with its own blocks.

Moving blocks from child to parent zones and back is fine. Moving the entire child block results in the error.

[Ive Created a REPL here](https://svelte.dev/repl/d61b7d27fd40426a8814549fce2dbe2b?version=3.41.0)

**Comment by isaacHagoel** [comment]
sorry. i missed it. have you figured it out or still an issue?

**Comment by awesomund** [comment]
I think I experienced a similar issue today, but luckily I also found a workaround.

I was using nested dnd-zones (with the same type). When dragging either the parent or child dnd-zone, i got a similar error:
`Uncaught TypeError: dzToConfig$1.get(...) is undefined`.

I looked at the [crazy nesting](https://svelte.dev/repl/fe8c9eca04f9417a94a8b6041df77139?version=3.42.1) example, and the only difference that really stood out from my own code was the filtering of the items with `SHADOW_PLACEHOLDER_ITEM_ID` as their id:

```
{#each node.items.filter(item => item.id !== SHADOW_PLACEHOLDER_ITEM_ID) as item(item.id)}
	<div animate:flip="{{duration: flipDurationMs}}" class="item">
		<svelte:self bind:nodes={nodes} node={nodes[item.id]} depth={depth-1}/>
	</div>
{/each}
```

After applying the same filter to my parent/child dnd-zones, i didn't see the error again. Adding the same filter to @mnashmi's example REPL also seems to work. My edit of the same code: https://svelte.dev/repl/d4c21245c79c48d28593e356a47a3c1f?version=3.42.1
